### PR TITLE
Disallow realm migration on synced realms

### PR DESCRIPTION
**Pull request by fronck**
## What, How & Why?

Now throws an error when client requests a realm that is both synced and has `deleteRealmIfMigrationNeeded` enabled.  These options are incompatible.

Added test that checks functionality.

This closes [#RJS-755](https://jira.mongodb.org/browse/RJS-755) / [HELP-18006](https://jira.mongodb.org/projects/HELP/queues/issue/HELP-18006).

## ☑️ ToDos

* [x] 📝 Changelog entry
~* [ ] 📝 `Compatibility` label is updated or copied from previous entry~
* [x] 🚦 Tests
~* [] 📝 Public documentation PR created or is not necessary~
~* [ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
~* [ ] typescript definitions file is updated~
* [x] jsdoc files updated
~* [ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ x.x.x Release notes (yyyy-MM-dd)
 
 ### Fixed
 * Fixed `create<T>(...)` deprecation warning. ([#3243](https://github.com/realm/realm-js/pull/3243))
+* Throw error when `deleteRealmIfMigrationNeeded` is requested on a synced realm (incompatible options)  ([RJS-755](https://jira.mongodb.org/browse/RJS-755))
```
Comment (kneth) [inline CHANGELOG.md:8]: Please remove reference to private Jira ticket.

**Review by kneth — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (fronck) [reply CHANGELOG.md]: Done.

File: tests/js/realm-tests.js [hunk 2]
```diff
@@ -1609,6 +1609,27 @@ module.exports = {
             });
     },
 
+    testNoMigrationOnSync: function() {
+        if (!global.enableSyncTests) {
+            return;
```
Comment (blagoev) [inline tests/js/realm-tests.js:1614]: should return a `Promise.resolve()` here. although we might have other tests doing that same thing I think we need to fix them as well

**Review by blagoev — APPROVED** [review]
with minor change in the test

File: tests/js/realm-tests.js (hunk 2 above)
Comment (fronck) [reply tests/js/realm-tests.js]: I've fixed it for this test, and created a ticket to check up on the rest of the tests.

**Pull request merged**